lib/rules/prop-types.js

```diff
@@ -261,7 +262,7 @@ module.exports = {
           // If it's a computed property, we can't make any further analysis, but is valid
           return key === '__COMPUTED_PROP__';
         }
-        if (propType === true) {
+        if (typeof propType === 'object' && Object.keys(propType).length === 0) {
```

Comment:
What is the typeof guarding against here? This allows `null`, for example.

Reply:
`null` would be checked in the if statement before:

```
if (!propType) {
    // If it's a computed property, we can't make any further analysis, but is valid
    return key === '__COMPUTED_PROP__';
}
```

Otherwise, if I remove the `typeof` condition, one test starts failing. Turns out `propType` in this case is a `function()` (not sure why) and it satifies the condition `Object.keys(propType).length === 0`. Perhaps should use https://www.npmjs.com/package/lodash.isplainobject ?

```
function Hello(props) {
  return <div>{props.name.constructor.firstname}</div>
}
Hello.propTypes = {
  name: PropTypes.shape({
    firstname: PropTypes.object
  })
};:

AssertionError: Should have 1 error but had 0: []
```

Reply:
Gotcha; makes sense.